libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/MapboxTripStarter.kt

```diff
@@ -0,0 +1,171 @@
+package com.mapbox.navigation.core.trip
+
+import android.annotation.SuppressLint
+import com.mapbox.android.core.permissions.PermissionsManager
+import com.mapbox.navigation.base.ExperimentalPreviewMapboxNavigationAPI
+import com.mapbox.navigation.core.MapboxNavigation
+import com.mapbox.navigation.core.lifecycle.MapboxNavigationApp
+import com.mapbox.navigation.core.lifecycle.MapboxNavigationObserver
+import com.mapbox.navigation.core.replay.route.ReplayRouteSession
+import com.mapbox.navigation.core.trip.MapboxTripStarterExtra.MAPBOX_TRIP_STARTER_FOLLOW_DEVICE
+import com.mapbox.navigation.core.trip.MapboxTripStarterExtra.MAPBOX_TRIP_STARTER_REPLAY_ROUTE
+import com.mapbox.navigation.core.trip.session.TripSessionState
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.SupervisorJob
+import kotlinx.coroutines.cancel
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.collect
+import kotlinx.coroutines.flow.update
+import kotlinx.coroutines.launch
+
+/**
+ * This makes it simpler to start a MapboxNavigation trip session. It can be used to enable replay
+ * or to follow the device location. The [MapboxTripStarter] is not able to observe when location
+ * permissions change, so you must notify it of changes. It will check location permissions upon
+ * attach. The location permissions are not required for replay sessions.
+ *
+ * In order to share the state between an App and Android Auto, it is recommended to make this
+ * class a singleton. That will be done automatically if you use [getRegisteredInstance].
+ */
+@ExperimentalPreviewMapboxNavigationAPI
+class MapboxTripStarter private constructor() : MapboxNavigationObserver {
+
+    private val optionsFlow = MutableStateFlow(MapboxTripStarterOptions.Builder().build())
+    private var replayRouteTripSession: ReplayRouteSession? = null
+    private var mapboxNavigation: MapboxNavigation? = null
+
+    private lateinit var coroutineScope: CoroutineScope
+
+    /**
+     * Signals that the [mapboxNavigation] instance is ready for use.
+     * @param mapboxNavigation
+     */
+    override fun onAttached(mapboxNavigation: MapboxNavigation) {
+        this.mapboxNavigation = mapboxNavigation
+        coroutineScope = CoroutineScope(SupervisorJob() + Dispatchers.Main.immediate)
+
+        // Initialize the options to be aware of the location permissions
+        val context = mapboxNavigation.navigationOptions.applicationContext
+        val granted = PermissionsManager.areLocationPermissionsGranted(context)
+        optionsFlow.update { it.toBuilder().isLocationPermissionGranted(granted).build() }
```

Comment:
Yeah I was thinking about this too because the `Options` for this are also kind of like a `State`. 

An approach I considered, was to make it a requirement for the developer to handle location permissions before setting the `FOLLOW_DEVICE` trip type. Then the location permission flag is deleted. My problem with that approach, is that it adds a requirement for the developer.

I'll explore a couple more ideas here. It seems often that these stand-alone objects will need to have `Options` and `State` flows.

Reply:
I've played with some ideas here, moved it to a new object. I made it a `*State` object because I'd like to follow a simple pattern for any services I make like this. Resolving here